contentcuration/contentcuration/viewsets/contentnode.py

```diff
@@ -264,21 +313,93 @@ def perform_bulk_update(self, serializer):
         serializer.save()
 
     def perform_bulk_create(self, serializer):
-        if ORPHAN_TREE_ID_CACHE_KEY not in cache:
-            tree_id = (
-                ContentNode.objects.filter(id=settings.ORPHANAGE_ROOT_ID)
-                .values_list("tree_id", flat=True)
-                .get()
-            )
-            # No reason for this to change so can cache for a long time
-            cache.set(ORPHAN_TREE_ID_CACHE_KEY, 24 * 60 * 60)
-        else:
-            tree_id = cache.get(ORPHAN_TREE_ID_CACHE_KEY)
         # Creating a new node, by default put it in the orphanage on initial creation.
         serializer.save(
-            tree_id=tree_id,
+            tree_id=get_orphan_tree_id(),
             parent_id=settings.ORPHANAGE_ROOT_ID,
             lft=1,
             rght=2,
             level=1,
         )
+
+    def copy(self, pk, user=None, from_key=None, **mods):
```

Comment:
Ah this is interesting. I was wondering where to put these types of functions too- maybe worth raising for the rest of the team?

Reply:
This pattern originated from @rtibbles I believe.

Reply:
Sounds good.